### ISSUE TITLE
Dict Key Literal Autocompletion does not work with Attributes

### ISSUE THREAD

**Issue by JakobDev** [microsoft/pylance-release#3687]
## Environment data

-   Language Server version: v2022.11.30
-   OS and version: Windows 10
-   Python version: 3.10.0

## Code Snippet

```python
from typing import Literal


test_literal = Literal[
    "Hello", 
    "World",
    "Test",
    "123"
]


a: dict[test_literal, str] = {}
# a[] Autocompletion of Keys works


def test_func() -> dict[test_literal, str]:
    pass


b = test_func()
# b[] Autocompletion of Keys works


class TestClass:
    def __init__(self) -> None:
        self.test_dict = dict[test_literal, str]

        # self.test_dict[] Autocompletion of Keys does not work


c = TestClass()
# c.test_dict[] Autocompletion of Keys does not work
```

## Repro Steps
Copy this code and start typing, where the comments are e.g. a[]. If you do this, the Autocompletion will pop-up and will provide Autocompletion for the Keys. ("Hello, World, Test, 123).

## Expected behavior
It should work everywhere

## Actual behavior
The autocompletion works only with a and b., but not with self.test_dict and c.test_dict.

## Logs

Logs have nothing to do with this

**Comment by debonte** [comment]
One issue is that `self.test_dict`, as written, is a type alias.

However, even if I change the declaration of `test_dict` to one of the following, and Pylance understands that it is of type `dict[test_literal, str]`, I still don't see the expected suggestions:

```python
self.test_dict = dict[test_literal, str]()
self.test_dict: dict[test_literal, str] = {}
```

**Comment by bschnurr** [comment]
Currently we only support a limited number of cases. 

This would be an enhancement. 
Thank you

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2022.11.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221141-30-November-2022-PreRelease)
